rust-runtime/aws-smithy-http-auth/src/auth.rs

```diff
@@ -39,42 +41,49 @@ impl From<AuthErrorKind> for AuthError {
 }
 
 /// Authentication configuration to connect to a Smithy Service
-#[derive(Clone, Debug, PartialEq)]
-pub struct AuthApiKey {
-    api_key: String,
+#[derive(Clone, Eq, PartialEq)]
+pub struct AuthApiKey(Arc<Inner>);
```

Comment:
Why `Arc` this?

Reply:
For the same reason specified here: https://github.com/awslabs/smithy-rs/blob/76ee00eb1737964ce240ae817f573980920f0607/aws/rust-runtime/aws-credential-types/src/credentials_impl.rs#L18-L19